pkg/frontend/admin_ovsflows_get.go

```diff
@@ -0,0 +1,195 @@
+		log.Error(err)
+	}
+
+	client := Client{config}
+	// Main logic written in f._getOSVFlow() func
+	b, err := client._getOVSFlows(ctx, r, log, config)
+	if err != nil {
+		log.Error(err)
+	}
+	reply(log, w, nil, b, err)
+}
+
+func (client Client) _getOVSFlows(ctx context.Context, r *http.Request, log *logrus.Entry, config *kubernetes.Clientset) ([]byte, error) {
+	//Checking if the node mentioned in the query is correct or not.
+	nodeName := r.URL.Query().Get("node")
+	if nodeName == "" {
+		err := api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, "", "Invalid Node Name")
+		return nil, err
+	}
+
+	// Checking if the given node exists
+	node, err := config.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Creating a pod which will run as privileged in the given node to execute "$ ovs-ofctl -O OpenFlow13 dump-flows br0" on that node to extract the OVS flows.
+	pod, err := client.createOVSFlowPod(ctx, node.Name, log)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	//  The below loop waits of the phase of the created pod to be Succeeded, if not it times out after 60 seconds and returns error.
+	for i := 0; pod.Status.Phase != corev1.PodSucceeded; i++ {
+		if i == 60 {
+			err = api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeInternalServerError, "", "")
+			return nil, err
+		}
+		pod, err = config.CoreV1().Pods(OVSFlowPodNamespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Error(err)
+			return nil, err
+		}
+		time.Sleep(time.Second)
+	}
+
+	// The OVS flows would be present in the logs of the created pod.
+	b, err := client.getPodLogs(ctx, log, pod)
+	if err != nil {
+		return nil, err
+	}
+
+	// Clearing the pod after extracting its logs.
+	defer func() {
+		config.CoreV1().Pods(OVSFlowPodNamespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+
+	return b, nil
+}
+
+type Client struct {
+	Clientset kubernetes.Interface
+}
+
+// This Function extract logs from the passed pod and return in the form of a slice bytes.
+func (c Client) getPodLogs(ctx context.Context, log *logrus.Entry, pod *corev1.Pod) ([]byte, error) {
+	opts := corev1.PodLogOptions{Container: OVSFlowContianerName}
+	r := c.Clientset.CoreV1().Pods(OVSFlowPodNamespace).GetLogs(pod.Name, &opts)
+	result, err := r.Do(ctx).Raw()
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+	return result, err
+}
+
+func (c Client) createOVSFlowPod(ctx context.Context, node string, log *logrus.Entry) (*corev1.Pod, error) {
+	var hostpathType corev1.HostPathType = "Directory"
+	hostPath := corev1.VolumeSource{HostPath: &corev1.HostPathVolumeSource{"/", &hostpathType}}
+	volumes := []corev1.Volume{{"host", hostPath}}
+	privileged := true
+	var runAsUser int64 = 0
+	i := 0
+	podName := OVSFlowPodName
+	for {
+		pod := &corev1.Pod{
+			metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
```

Comment:
This can be ommited. client-go should set it automatically given that scheme set correctly (should ube ok, see for refrence `pkg/util/scheme/scheme.go`)

Reply:
Sure ... I will check that ..